c7n/registry.py

```diff
@@ -45,28 +45,47 @@ class PluginRegistry(object):
 
     """
 
+    event_times = ('register', 'unregister', 'final')
+
     def __init__(self, plugin_type):
         self.plugin_type = plugin_type
         self._factories = {}
+        self._events = {x: [] for x in self.event_times}
+
+    def add_event(self, time, func):
```

Comment:
subscribe is a bit more clear, also time event_name/event_type.. time is ambiguous.